dynamo-future-file-extension/extension/Extension.cs

```diff
@@ -0,0 +1,128 @@
+            {
+                harmony.Patch(originalOpenFileFromPath, new HarmonyMethod(prefix));
+            }
+        }
+        public static bool Prefix(DynamoModel __instance, string filePath, bool forceManualExecutionMode = false)
+        {
+
+            __instance?.Logger?.Log($"openfilepath has been patched from{Assembly.GetExecutingAssembly().FullName }");
+            var knownInputTypeNames = Enum.GetValues(typeof(NodeInputTypes)).Cast<NodeInputTypes>().Select(x => ToEnumString(x));
+            XmlDocument xmlDoc;
+            Exception ex;
+            if (DynamoUtilities.PathHelper.isValidXML(filePath, out xmlDoc, out ex))
+            {
+                openxmlmethod.Invoke(__instance, new object[] { xmlDoc, filePath, forceManualExecutionMode });
+                return false;
+            }
+            else
+            {
+                // These kind of exceptions indicate that file is not accessible 
+                if (ex is IOException || ex is UnauthorizedAccessException)
+                {
+                    throw ex;
+                }
+                if (ex is System.Xml.XmlException)
+                {
+                    // XML opening failure can indicate that this file is corrupted XML or Json
+                    string fileContents;
+                    if (DynamoUtilities.PathHelper.isValidJson(filePath, out fileContents, out ex))
+                    {
+                        //at this point filecontents contains a json string - 
+                        //lets parse it, do our transformation and pass it to open method.
+
+                        // TODO we could do more complex replacements, for example, we could replace the WorkspaceConverter.ReadJson method 
+                        // and completely alter how nodeInputs are deserialized. This is just a POC to fix the issue with minimum effort.
+
+                        var root = JObject.Parse(fileContents);
+                        //find all inputs that don't exist in the known types - these are the ones we can't deserialize correctly.
+                        //TODO reason about case sensitivity.
+                        var matchingTokens = root.SelectTokens("Inputs[*].Type").Where(x => !knownInputTypeNames.Contains(x.Value<string>()));
+                        //for now we just replace with "Selection" - but we could avoid deserializing this entire input.
+                        matchingTokens.ToList().ForEach(x => x.Replace(JToken.FromObject("Selection")));
+                        fileContents = root.ToString();
+
+                        openJsonFileMethod.Invoke(__instance, new object[] { fileContents, filePath, forceManualExecutionMode });
+                        return false;
+                    }
+                    else
+                    {
+                        throw ex;
+                    }
+                }
+            }
+            return false;
```

Comment:
maybe return true here...
Returning false means the original method gets skipped.

Reply:
yes, I want to skip the original method entirely.

Reply:
So it looks like this line of code will silently stop the loading of a graph.
Shouldn't we log something to notify the user of the reason for the failure ?

Reply:
yes, if I stick with this thats a good idea, I actually had a comment saying the same thing and erased it for some reason 😢